File: unity_project/Assets/Scripts/Core/GameConfig.cs
using UnityEngine;

namespace Core {
    public static class GameConfig {
        public const float GravityConstant = 300.0f;
        public const float GravityRadiusScale = 3.0f;
        public const float ThrustPower = 300.0f;
        public const float MinPlanetRadius = 15.0f;
        public const float MaxPlanetRadius = 50.0f;
        public const int MinPlanetsPerChunk = 2;
        public const int MaxPlanetsPerChunk = 5;
        public const float ChunkSize = 600.0f;
        public const float ShipCollisionRadius = 5.0f;
        public const float FuelSpawnInterval = 10000.0f;
        public const int EngineType = 2;
        public const int MusicType = 2;
        public const float EngineVolume = 0.3f;
        public const float MusicVolume = 2.5f;
        public const float StarFuelBurnRate = 20.0f;
        public const float NearMissSpeedThreshold = 300.0f;
        public const float NearMissFuelReward = 10.0f;
        public const float NearMissCooldown = 1.0f;
        public const float NearMissDistance = 40.0f;
        public const float DebugStartDistance = 0.0f;
        public const float LandingMaxSpeed = 65.0f;
        public const int DebugShowNearMiss = 0;
    }
}

File: python_project/game_config.py
import json
import os

class GameConfig:
    gravityConstant = 300
    gravityRadiusScale = 3.0
    thrustPower = 300
    minPlanetRadius = 15
    maxPlanetRadius = 50
    minPlanetsPerChunk = 2
    maxPlanetsPerChunk = 5
    chunkSize = 600
    shipCollisionRadius = 5
    fuelSpawnInterval = 10000
    engineType = 2
    musicType = 2
    engineVolume = 0.3
    musicVolume = 2.5
    starFuelBurnRate = 20.0
    nearMissSpeedThreshold = 300
    nearMissFuelReward = 10.0
    nearMissCooldown = 1.0
    nearMissDistance = 40
    debugStartDistance = 0
    landingMaxSpeed = 65.0
    debugShowNearMiss = 0

    @classmethod
    def load(cls, filepath="../game_config.json"):
        if os.path.exists(filepath):
            with open(filepath, 'r') as f:
                data = json.load(f)
                for k, v in data.items():
                    if hasattr(cls, k):
                        setattr(cls, k, v)
        else:
            print(f"Warning: Configuration file {filepath} not found. Using defaults.")

# Try to load the config relative to the script
script_dir = os.path.dirname(os.path.abspath(__file__))
config_path = os.path.join(script_dir, "..", "game_config.json")
GameConfig.load(config_path)
